packages/blockchain/src/db/helpers.ts

```diff
@@ -43,14 +43,7 @@ function DBSetBlockOrHeader(blockBody: Block | BlockHeader): DBOp[] {
 
   const isGenesis = header.number === BIGINT_0
 
-  if (
```

Comment:
I think the optimization of not saving the body when there aren't non zero fields is not worth it compared to the body load complexity where we would push empty `[]` to resconstruct the body